File: Python/Q9_Interfaces.py
from abc import ABC, abstractmethod

class Imprimível(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimível):
    def __init__(self, conteudo):
        self.conteudo = conteudo
    def imprimir(self):
        print(f"Imprimindo Relatório: {self.conteudo}")

class Contrato(Imprimível):
    def __init__(self, conteudo):
        self.conteudo = conteudo
    def imprimir(self):
        print(f"Imprimindo Contrato: {self.conteudo}")

relatorio = Relatorio("Relatorio")
contrato = Contrato("Contrato")
relatorio.imprimir()
contrato.imprimir()

File: go/Q9_Interfaces/Q9_Interfaces.go
package main

import (
	"fmt"
)

type Imprimível interface {
	Imprimir()
}

type Relatorio struct {
	conteudo string
}

func (r Relatorio) Imprimir() {
	fmt.Printf("Imprimindo relatório: %s\n", r.conteudo)
}

type Contrato struct {
	conteudo string
}

func (c Contrato) Imprimir() {
	fmt.Printf("Imprimindo contrato: %s\n", c.conteudo)
}


func main() {
	relatorio := Relatorio{conteudo: "Relatório"}

	contrato := Contrato{conteudo: "Contrato"}

	relatorio.Imprimir()
	contrato.Imprimir()
}
